src/mono/netcore/System.Private.CoreLib/src/System/Environment.Unix.Mono.cs

```diff
@@ -19,17 +19,26 @@ static string GetEnvironmentVariableCore (string variable)
 		{
 			Debug.Assert(variable != null);
 
+			variable = TrimStringOnFirstZero (variable);
+
+			// call getenv directly if s_environment is not yet initialized
 			if (s_environment == null) {
 				using (var h = RuntimeMarshal.MarshalString (variable)) {
 					return internalGetEnvironmentVariable_native (h.Value);
 				}
 			}
 
-			variable = TrimStringOnFirstZero (variable);
+			string value = "";
 			lock (s_environment) {
-				s_environment.TryGetValue (variable, out string value);
-				return value;
+				if (!s_environment.TryGetValue (variable, out value)) {
+					// on some platform s_environment can be empty after EnsureEnvironmentCached (), e.g. iOS
+					using (var h = RuntimeMarshal.MarshalString (variable)) {
```

Comment:
Nits -- this repo requires 4 space not tab indentation please. https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md

Also we avoid spaces between method and opening paren `s_environment.TryGetValue (variable, out value)` --> `s_environment.TryGetValue(variable, out value)` etc.

None of these matter on their own but we probably want to try to avoid 2 different styles in the repo.

Reply:
@danmosemsft it matches our mono/mono guidelines. When we create new files here in dotnet/runtime we follow dotnet/runtime. But not sure what to do with the existing files we copied from mono/mono. Code-formatting to the new guidelines for all netcore/*.cs will make git blame unusable I guess.

Reply:
Losing some git blame is unfortunate -- but that often gets lost in refactors anyway. It's not too difficult to sync back and do git blame from that point. I don't think we want folks to have to configure tabs vs spaces when they go from file to file. 😈

Reply:
cc @akoeplinger ^

Reply:
I think we copied the files without history anyway so sooner we can do it less history we lose.

Reply:
Yes these files in src/mono/netcore aren't synced so we can bring them up to standards.

I agree we should reformat but please not in this PR :)

Reply:
(fixed)